src/rawposix/src/sys_calls.rs

```diff
@@ -0,0 +1,536 @@
+    println!("[rawposix|waitpid] cp-3");
+    // return child's cageid
+    zombie.cageid as i32
+}
+
+/// Reference to Linux: https://man7.org/linux/man-pages/man2/wait.2.html
+///
+/// See comments of waitpid_syscall
+pub fn wait_syscall(
+    cageid: u64,
+    status_arg: u64,
+    status_cageid: u64,
+    arg2: u64,
+    arg2_cageid: u64,
+    arg3: u64,
+    arg3_cageid: u64,
+    arg4: u64,
+    arg4_cageid: u64,
+    arg5: u64,
+    arg5_cageid: u64,
+    arg6: u64,
+    arg6_cageid: u64,
+) -> i32 {
+    // would sometimes check, sometimes be a no-op depending on the compiler settings
+    if !(sc_unusedarg(arg2, arg2_cageid)
+        && sc_unusedarg(arg3, arg3_cageid)
+        && sc_unusedarg(arg4, arg4_cageid)
+        && sc_unusedarg(arg5, arg5_cageid)
+        && sc_unusedarg(arg6, arg6_cageid))
+    {
+        return syscall_error(Errno::EFAULT, "waitpid", "Invalid Arguments");
+    }
+    // left type conversion done inside waitpid_syscall
+    waitpid_syscall(
+        cageid,
+        0,
+        0,
+        status_arg,
+        status_cageid,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+    )
+}
+
+pub fn getpid_syscall(
+    cageid: u64,
+    arg1: u64,
+    arg1_cageid: u64,
+    arg2: u64,
+    arg2_cageid: u64,
+    arg3: u64,
+    arg3_cageid: u64,
+    arg4: u64,
+    arg4_cageid: u64,
+    arg5: u64,
+    arg5_cageid: u64,
+    arg6: u64,
+    arg6_cageid: u64,
+) -> i32 {
+    // would sometimes check, sometimes be a no-op depending on the compiler settings
+    if !(sc_unusedarg(arg1, arg1_cageid)
+        && sc_unusedarg(arg2, arg2_cageid)
+        && sc_unusedarg(arg3, arg3_cageid)
+        && sc_unusedarg(arg4, arg4_cageid)
+        && sc_unusedarg(arg5, arg5_cageid)
+        && sc_unusedarg(arg6, arg6_cageid))
+    {
+        return syscall_error(Errno::EFAULT, "exec", "Invalide Cage ID");
+    }
+
+    let cage = get_cage(cageid).unwrap();
+
+    return cage.cageid as i32;
+}
+
+pub fn getppid_syscall(
+    cageid: u64,
+    arg1: u64,
+    arg1_cageid: u64,
+    arg2: u64,
+    arg2_cageid: u64,
+    arg3: u64,
+    arg3_cageid: u64,
+    arg4: u64,
+    arg4_cageid: u64,
+    arg5: u64,
+    arg5_cageid: u64,
+    arg6: u64,
+    arg6_cageid: u64,
+) -> i32 {
+    // would sometimes check, sometimes be a no-op depending on the compiler settings
+    if !(sc_unusedarg(arg1, arg1_cageid)
+        && sc_unusedarg(arg2, arg2_cageid)
+        && sc_unusedarg(arg3, arg3_cageid)
+        && sc_unusedarg(arg4, arg4_cageid)
+        && sc_unusedarg(arg5, arg5_cageid)
+        && sc_unusedarg(arg6, arg6_cageid))
+    {
+        return syscall_error(Errno::EFAULT, "exec", "Invalide Cage ID");
+    }
+
+    let cage = get_cage(cageid).unwrap();
+
+    return cage.parent as i32;
+}
+
+/// Those functions are required by wasmtime to create the first cage. `verbosity` indicates whether
+/// detailed error messages will be printed if set
+pub fn lindrustinit(verbosity: isize) {
```

Comment:
These two function names are pretty bad (and they're my fault!) as relics from the really initial version of rustposix. maybe we change this to rawposix_initialize etc? makes it a bit better.

Reply:
they're also not really "sys system calls" is there a better file for them?